File: Week_06/518/coin_change_2.go
package main

func change(amount int, coins []int) int {
	dp := make([]int, amount+1)
	dp[0] = 1

	for _, c := range coins {
		for i := c; i <= amount; i++ {
			dp[i] += dp[i-c]
		}
	}
	return dp[amount]
}

File: Week_06/518/coin_change_2.py
from typing import List

class Solution:
    def change(self, amount: int, coins: List[int]) -> int:
        if not coins or not amount:
            return 0
        dp = [0] * (amount+1)
        dp[0] = 1

        for i in range(1, amount+1):
            for c in coins:
                if i >= c:
                    dp[i] += dp[i-c]
        return dp[amount]